### PR TITLE
Allow non-existent atoms in L16

### PR DESCRIPTION
**Pull request by richardicooper**
Allows L16 to be input before L5. Allows atom renaming without breaking whole refinement and L16 storage.
They may be not-yet existent. Restraint is still ignored and warning is printed during SFLS.

### CODE REVIEW

File: crystals/lexical_helper.F90 [hunk 1]
```diff
@@ -1628,7 +1628,42 @@ subroutine check_atom(image_text, ierror)
...
+                                                  'UXZQIV  ', &
+                                                  'UYYQIV  ', &
+                                                  'UYZQIV  ', &
+                                                  'UZZQIV  ', &
+                                                  'UEIG    ', &
+                                                  'UQISO   ', &
+                                                  'UVOL    ', &
+                                                  'UEQIV   ', &
+                                                  'UTLS    ', &
+                                                  'XCHIV   ', &
```
Comment (pascal22p) [inline crystals/lexical_helper.F90]: Is it not pretty much everything? That would render the subroutine check_atom useless

**Comment by richardicooper** [comment]
Is it necessary to check atom existence at the parser stage? The 2018
change ends up sometimes wiping L16 completely when the user has a few bad
restraints in their L16 due to the error condition. Could we alter the
parser to give a warning only?

What is the benefit of checking atoms at the input stage?

**Comment by pascal22p** [comment]
It is not really necessary. I think it was to avoid a not very explicit error later on.
Is it not because it tells you what restraint has an error but not what is wrong in the restraint?

To kill the check, just remove the line 247, 248.

**Comment by richardicooper** [comment]
OK - I've kept the check, but it no longer causes an error condition on list input. It does print the
warning message, because it's useful.

**Comment by pascal22p** [comment]
take out line 248. When checking for any errors during checking it returns if some errors are found. If you let it continue it should be find. you may need to reset ierror to zero after the call to check_atom.

You might consider having a flag set for syntax checking depending on what you are doing to avoid flooding the user with invalid atoms he should not have to worry.

**Comment by richardicooper** [comment]
This will fail lexical test because of changed logic. Test output updates to follow.

**Comment by richardicooper** [comment]
> take out line 248. 

This is the approach I've taken + changed the word Error to Warning in the check_atoms function.

> you may need to reset ierror to zero after the call to check_atom.

It's reset in each function so no worries.

> You might consider having a flag set for syntax checking depending on what you are 
> doing to avoid flooding the user with invalid atoms he should not have to worry.

This is where knowing the instruction context of the line being parsed would be useful. I'll leave it as it is for now.

**Pull request merged**